### DISCUSSION TITLE
How to filter dashboard on a nested JSON value?

### DISCUSSION THREAD

**Discussion by glossyio** [thingsboard/thingsboard#11953] [Q&A]
I feel like I'm missing something obvious here, but I can't figure it out. I'd appreciate the help!

**What am I trying to do?**
I am trying to create a dashboard that contains Hourly counts for the Current Day of a `event` telemetry object, and I want some of the charts filtered by particular values from nested keys; e.g. filter where the key `event`.`entered_zone` CONTAINS the words "zone_capture".  However, when  I create and apply a filter to chart, it gives back No Data.  Note: At this point I don't need to parse the JSON value, just need to check if a particular value exists, so I'm only using the CONTAINS operation. I thought that would be easiest to start, but I'm still not getting it.

How do I filter the chart by [any of] the nested JSON values for my time series data `event`?

**What have I tried?**
- Watched  https://thingsboard.io/docs/user-guide/advanced-filters/
- Read up on https://github.com/thingsboard/thingsboard/issues/6473 
- Read up on https://github.com/thingsboard/thingsboard/issues/7675

This is the dashboard:  Bottom Right chart is the Filtered one displaying No Data.  It should be a subset of the Bottom Left chart:
![image](https://github.com/user-attachments/assets/222ae6cc-ef56-4971-a0fb-bc9a264304f1)

Filter (that doesn't work).  I tried playing around with the Key Type and changing the key name to event.entered_zone and just event to no avail:
![image](https://github.com/user-attachments/assets/5223950e-f048-43ac-b804-929afece2f23)

I even queried the backend PostgreSQL database to see how the data was stored, and it looks like it's properly storing it as a JSON object; i.e. in the `json_v` column for `key=90` which is the `event` telemetry key.

![image](https://github.com/user-attachments/assets/02562f92-6be9-486e-8b25-599fea9d64d8)

Sample payload that I send to ThingsBoard, for reference:

```
{
    "ts": 1729784007420.6409,
    "values": {
        "event": {
            "id": "1729783982.718412-rl2b6l",
            "camera": "picam_h264",
            "label": "person",
            "sub_label": null,
            "top_score": 0.84375,
            "frame_time": 1729784007.420641,
            "start_time": 1729783982.718412,
            "end_time": 1729784012.555179,
            "entered_zones": "[\"zone_capture\",\"zone_near\",\"zone_far\"]",
            "score": 0.68359375,
            "area": 4187,
            "ratio": 0.6708860759493671,
            "motionless_count": 1,
            "position_changes": 1,
            "attributes": "{}",
            "direction_calc": "outbound",
            "deployment_id": "1727678532209-4110962f-d",
            "source": "frigate"
        }
    }
}
```

**Environment**
- Docker ThingsBoard v.3.7.0 CE with PostgreSQL

**Comment by icoco** [comment]
solved ?  face same question

**Reply by glossyio** [reply]
I have not gotten a solution on this yet. I assume I will have to do JSON parsing to flatten the structure, and I have not yet bit off that work.
